pkg/loki/modules.go

```diff
@@ -463,6 +463,16 @@ func (t *Loki) initQueryFrontend() (_ services.Service, err error) {
 	t.Server.HTTP.Handle("/loki/api/v1/tail", defaultHandler)
 	t.Server.HTTP.Handle("/api/prom/tail", defaultHandler)
 
+	if t.frontend == nil {
+		return services.NewIdleService(nil, func(_ error) error {
+			if t.stopper != nil {
+				t.stopper.Stop()
+				t.stopper = nil
```

Comment:
not quite sure why we set the stopper to nil after stopping, but doesn't look harmful :)

Reply:
To not stop twice. Cortex does that I figured let's do it too. Not sure it's required.